conformance/lang/expressions/logical-expr/logical_and.balt

```diff
@@ -0,0 +1,699 @@
+    io:println(false && 2 is int); // @output false
+    io:println(2 is int && false); // @output false
+    io:println(2 is int && 2 is int); // @output true
+}
+
+Test-Case: output
+Description: Test logical expression using equality expression.
+Labels: logical-expr, equality
+
+function init() {
+    io:println(true && 2 == 3); // @output false
+    io:println(3 == 3 && true); // @output true
+    io:println(false && 3 == 3); // @output false
+    io:println(2 == 3 && false); // @output false
+    io:println(2 == 3 && 3 == 3); // @output false
+
+    io:println(true && 2 != 3); // @output true
+    io:println(3 != 3 && true); // @output false
+    io:println(false && 3 != 3); // @output false
+    io:println(2 != 3 && false); // @output false
+    io:println(2 != 3 && 3 != 3); // @output false
+
+    io:println(true && 2 === 3); // @output false
+    io:println(3 === 3 && true); // @output true
+    io:println(false && 3 === 3); // @output false
+    io:println(2 === 3 && false); // @output false
+    io:println(2 === 3 && 3 === 3); // @output false
+
+    io:println(true && 2 !== 3); // @output true
+    io:println(3 !== 3 && true); // @output false
+    io:println(false && 3 !== 3); // @output false
+    io:println(2 !== 3 && false); // @output false
+    io:println(2 !== 3 && 3 !== 3); // @output false
+}
+
+Test-Case: output
+Description: Test logical expression with logical expression.
+Labels: logical-expr
+
+function init() {
+    io:println(true && true && false); // @output false
+    io:println(true && false && true); // @output false
+    io:println(false && true && false); // @output false
+    io:println(true && false && false); // @output false
+    io:println(true && false && true && false); // @output false
+}
+
+Test-Case: output
+Description: Test evaluation order of logical expression.
+Labels: logical-expr, function-call-expr, additive-expr, is-expr
+
+int falseEval = 0;
+int trueEval = 0;
+
+function returnTrueVal() returns boolean {
+    trueEval = trueEval + 1;
+    return true;
+}
+
+function returnFalseVal() returns boolean {
+    falseEval = falseEval + 1;
+    return false;
+}
+
+function resetEvalIndicators() {
+    falseEval = 0;
+    trueEval = 0;
+}
+
+function init() {
+    boolean a = returnFalseVal() && returnTrueVal();  
+    io:println(falseEval); // @output 1
+    io:println(trueEval); // @output 0
+    io:println(a is false); // @output true
+
+    resetEvalIndicators();
+    a = returnTrueVal() && returnFalseVal();    
+    io:println(falseEval); // @output 1
+    io:println(trueEval); // @output 1
+    io:println(a is false); // @output true
+
+    resetEvalIndicators();
+    a = returnTrueVal() && returnTrueVal();    
+    io:println(trueEval); // @output 2
+    io:println(a is true); // @output true
+
+    resetEvalIndicators();
+    a = returnFalseVal() && returnFalseVal();
+    io:println(falseEval); // @output 1
+    io:println(a is false); // @output true
+}
```

Comment:
Shall we add a test to cover https://github.com/ballerina-platform/ballerina-lang/issues/35693 also?
The following also doesn't work.
```ballerina
type TRUE true;

type FALSE false;

function init() {
    TRUE a = true;
    FALSE b = false;
    boolean c = a && b;  // operator '&&' not defined for 'TRUE' and 'FALSE'
}
```